--- a/targets/mediatek-filogic
+++ b/targets/mediatek-filogic
@@ -39,6 +39,10 @@
 	factory = false,
 })
 
+device('cudy-ap3000-v1', 'cudy_ap3000-v1', {
+	factory = false,
+})
+
 device('cudy-m3000-v1', 'cudy_m3000-v1', {
 	factory = false,
 })
@@ -47,6 +51,10 @@
 	factory = false,
 })
 
+device('cudy-tr3000-256mb-v1', 'cudy_tr3000-256mb-v1', {
+	factory = false,
+})
+
 device('cudy-tr3000-v1', 'cudy_tr3000-v1', {
 	factory = false,
 })
