--- a/target/linux/mediatek/filogic/base-files/etc/init.d/bootcount
+++ b/target/linux/mediatek/filogic/base-files/etc/init.d/bootcount
@@ -5,13 +5,6 @@
 
 boot() {
 	case $(board_name) in
-	dlink,aquila-pro-ai-m30-a1)
-		if grep -q bootpart=ubi0 /proc/cmdline; then
-			fw_setenv bootpart 0
-		else
-			fw_setenv bootpart 1
-		fi
-		;;
 	zyxel,ex5700-telenor)
 		fw_setenv uboot_bootcount 0
 		;;
--- a/target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh
+++ b/target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh
@@ -116,6 +116,10 @@
 	yuncore,ax835)
 		default_do_upgrade "$1"
 		;;
+	dlink,aquila-pro-ai-m30-a1)
+		fw_setenv sw_tryactive 0
+		nand_do_upgrade "$1"
+		;;
 	glinet,gl-mt6000)
 		CI_KERNPART="kernel"
 		CI_ROOTPART="rootfs"
