diff --git a/package/base-files/files/lib/upgrade/fwtool.sh b/package/base-files/files/lib/upgrade/fwtool.sh
index 8bd00a3332..1b3d588bcd 100644
--- a/package/base-files/files/lib/upgrade/fwtool.sh
+++ b/package/base-files/files/lib/upgrade/fwtool.sh
@@ -70,9 +70,9 @@ fwtool_check_image() {
                        fi

                        # minor compat version -> sysupgrade with -n required
-                       if [ "${devicecompat#.*}" != "${imagecompat#.*}" ] && [ "$SAVE_CONFIG" = "1" ]; then
+                       if [ "${devicecompat#.*}" != "${imagecompat#.*}" ] && [ "$SAVE_CONFIG" = "1" ] &&  [ "$device" != "xiaomi,mi-router-4a-gigabit" ] && [ "$device" != "ubnt-erx" ] ; then
                                [ "$IGNORE_MINOR_COMPAT" = 1 ] && return 0
-                               v "The device is supported, but the config is incompatible to the new image ($devicecompat->$imagecompat). Please upgrade without keeping config (sysupgrade -n)."
+                               v "The device $device is supported, but the config is incompatible to the new image ($devicecompat->$imagecompat). Please upgrade without keeping config (sysupgrade -n)."
                                [ -n "$compatmessage" ] && v "$compatmessage"
                                return 1
                        fi
diff --git a/target/linux/ramips/image/mt7621.mk b/target/linux/ramips/image/mt7621.mk
index b2e3285194..5976db362c 100644
--- a/target/linux/ramips/image/mt7621.mk
+++ b/target/linux/ramips/image/mt7621.mk
@@ -2763,7 +2763,6 @@ endef
 TARGET_DEVICES += xiaomi_mi-router-4

 define Device/xiaomi_mi-router-4a-gigabit
-  $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   IMAGE_SIZE := 14848k
   DEVICE_VENDOR := Xiaomi
